Sources/Operators/WithLatestFrom.swift

```diff
@@ -36,6 +36,43 @@ public extension Publisher {
       -> Publishers.WithLatestFrom<Self, Other, Other.Output> {
     return .init(upstream: self, second: other) { $1 }
   }
+
+  /// Upon an emission from self, emit the latest value from the
+  /// second and third publisher, if any exists.
+  ///
+  /// - parameter other: A second publisher source.
+  /// - parameter other1: A third publisher source.
+  ///
+  /// - returns: A publisher containing the latest value from the second and third publisher, if any.
+  func withLatestFrom<Other, Other1>(_ other: Other,
+                                     _ other1: Other1)
+    -> AnyPublisher<(Self.Output, Other.Output, Other1.Output), Failure>
+    where Other: Publisher, Other1: Publisher, Other.Failure == Failure, Other1.Failure == Failure {
+      let combined = combineLatest(other, other1)
+        .eraseToAnyPublisher()
+      return withLatestFrom(combined)
+        .eraseToAnyPublisher()
+    }
```

Comment:
I think we can go down this road:

```suggestion
  /// Upon an emission from self, emit the latest value from the
  /// second and third publisher, if any exists.
  ///
  /// - parameter other: A second publisher source.
  /// - parameter other1: A third publisher source.
  ///
  /// - returns: A publisher containing the latest value from the second and third publisher, if any.
  func withLatestFrom<Other: Publisher, Other1: Publisher>(_ other: Other,
                                                           _ other1: Other1)
      -> Publishers.WithLatestFrom<Self,
                                   AnyPublisher<(Other.Output, Other1.Output), Self.Failure>,
                                   (Self.Output, Other.Output, Other1.Output)>
      where Other.Failure == Failure, Other1.Failure == Failure {
          let combined = other.combineLatest(other1)
              .eraseToAnyPublisher()

          return withLatestFrom(combined, resultSelector: { ($0, $1.0, $1.1) })
  }
```